src/app/pages/system/general-settings/support/support.component.ts

```diff
@@ -0,0 +1,207 @@
+              {}
+
+  ngOnInit() {
+    this.ws.call('system.info').subscribe((res) => {
+      this.systemInfo = res;
+      this.systemInfo.memory = (res.physmem / 1024 / 1024 / 1024).toFixed(0) + ' GiB';
+      if (res.system_product.includes('MINI')) {
+        this.getMiniImage(res.system_product)
+      } else {
+        this.getServerImage(res.system_product);
+      };
+      if (res.license) {
+        this.hasLicense = true;
+        this.licenseInfo = res.license;
+        this.parseLicenseInfo();
+        this.ws.call('support.is_available').subscribe(res => {
+          if (res) {
+            this.ws.call('support.is_available_and_enabled').subscribe(res => {
+            })
+          }
+        })
+      }
+      this.licenseButtonText = this.hasLicense ? helptext.updateTxt : helptext.enterTxt;
+    });
+    setTimeout(() => {
+      this.ws.call('truenas.is_production').subscribe((res) => {
+        this.isProduction = res;
+      });
+    }, 500)
+  };
+
+  parseLicenseInfo() {
+    this.licenseInfo.features.length === 0 ? this.licenseInfo.featuresString = 'NONE' : 
+      this.licenseInfo.featuresString = this.licenseInfo.features.join(', ');
+    let expDateConverted = new Date(this.licenseInfo.contract_end.$value);
+    this.licenseInfo.expiration_date = this.licenseInfo.contract_end.$value;
+    this.systemInfo.system_serial_ha ?
+        this.systemInfo.serial = this.systemInfo.system_serial + ' / ' + this.systemInfo.system_serial_ha :
+        this.systemInfo.serial = this.systemInfo.system_serial;
+    this.licenseInfo.addhw.length === 0 ? this.licenseInfo.add_hardware = 'NONE' : 
+      this.licenseInfo.add_hardware = this.licenseInfo.addhw.join(', ');
+    const now = new Date(this.systemInfo.datetime.$date);
+    const then = expDateConverted;
+    this.licenseInfo.daysLeftinContract = this.daysTillExpiration(now, then);
+  }
+
+  daysTillExpiration(now, then) {
+    const oneDay = 24*60*60*1000; // milliseconds in a day
+    return Math.round((then.getTime() - now.getTime())/(oneDay))
+  }
+
+  getServerImage(sys_product) {
+    let imagePath = '';
+    this.serverList.forEach(model => {
+      if (sys_product.includes(model)) {
+        imagePath = `/servers/${model}.png`;
+      }
+    })
+    if (imagePath) {
+      this.product_image = imagePath;
+    } else {
+      this.product_image = 'ix-original-cropped.png';
+      this.isProductImageTall = true;
+      this.extraMargin = false;
+    }
+  }
+
+  getMiniImage(sys_product) {
+    switch(sys_product){
+      case "FREENAS-MINI-2.0":
+      case "FREENAS-MINI-3.0-E":
+      case "FREENAS-MINI-3.0-E+":
+        this.product_image = 'freenas_mini_cropped.png';
+      break;
+      case "FREENAS-MINI-3.0-X":
+      case "FREENAS-MINI-3.0-X+":
+        this.product_image = 'freenas_mini_x_cropped.png';
+      break;
+      case "FREENAS-MINI-XL":
+      case "FREENAS-MINI-3.0-XL+":
+        this.product_image = 'freenas_mini_xl_cropped.png';
```

Comment:
Does this need to be updated to TrueNAS Mini or be able to handle either naming?

Reply:
Will the new model names/numbers be reported  by middleware (system.info) the same as above, except with TRUENAS in place of FREENAS?

Reply:
We need both for 3.0-*
FREENAS and TRUENAS

Reply:
i think this changed was made in 12.0 and perhaps missed a backport to master?